pom.xml

```diff
@@ -163,6 +163,10 @@
     <scala.version>2.12.15</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
     <scalatest-maven-plugin.version>2.0.2</scalatest-maven-plugin.version>
+    <!-- SPARK-36547: This needs to be managed in different profiles to avoid
+      errors building different Hadoop versions -->
+
+    <scala-maven-plugin.version>4.4.0</scala-maven-plugin.version>
```

Comment:
Why not 4.5.6 ?
https://search.maven.org/artifact/net.alchim31.maven/scala-maven-plugin/4.5.6/maven-plugin

Reply:
i reverted to the one which was known to work on spark and hadoop 3, which was enough to unblock me without me having to go near any futher regressions caused by later versions of something which is clearly brittle to changes in *something* (transient dependencies?)

Reply:
OK. I will try it myself!